### PR TITLE
Surface more information about task status in tests

### PR DESCRIPTION
**Pull request by abhishekagarwal87**
I see some test runs failing because task status is not as expected. It will be helpful to know what error the task has.

### CODE REVIEW

File: indexing-service/src/test/java/org/apache/druid/indexing/worker/shuffle/LocalIntermediaryDataManagerAutoCleanupTest.java [hunk 1]
```diff
@@ -144,8 +146,8 @@ private boolean isCleanedUpAfter2s(String supervisorTaskId, Period timeoutPeriod
     // Start the data manager and the cleanup cycle
     intermediaryDataManager.start();
 
-    // Check the state of the partition after 2s
-    Thread.sleep(2000);
+    // Check the state of the partition after 3s
+    Thread.sleep(3000);
```
Comment (kfaraz) [inline indexing-service/src/test/java/org/apache/druid/indexing/worker/shuffle/LocalIntermediaryDataManagerAutoCleanupTest.java:150]: Thanks for fixing this, @abhishekagarwal87 . I guess the 2s was causing some flakiness if the cleanup tick was missed.

File: indexing-service/src/test/java/org/apache/druid/indexing/common/task/batch/parallel/AbstractMultiPhaseParallelIndexingTest.java [hunk 2]
```diff
@@ -176,7 +176,7 @@ void runTaskAndVerifyStatus(Task task, TaskState expectedTaskStatus)
   {
     task.addToContext(Tasks.FORCE_TIME_CHUNK_LOCK_KEY, lockGranularity == LockGranularity.TIME_CHUNK);
     TaskStatus taskStatus = getIndexingServiceClient().runAndWait(task);
-    Assert.assertEquals(expectedTaskStatus, taskStatus.getStatusCode());
+    Assert.assertEquals(taskStatus.toString(), expectedTaskStatus, taskStatus.getStatusCode());
```
Comment (kfaraz) [inline indexing-service/src/test/java/org/apache/druid/indexing/common/task/batch/parallel/AbstractMultiPhaseParallelIndexingTest.java:179]: Maybe a longer message like: "Expected task status to be ..."

**Review by kfaraz — APPROVED** [review]
Minor comment. Overall LGTM.

File: indexing-service/src/test/java/org/apache/druid/indexing/common/task/batch/parallel/AbstractMultiPhaseParallelIndexingTest.java (hunk 2 above)
Comment (abhishekagarwal87) [reply indexing-service/src/test/java/org/apache/druid/indexing/common/task/batch/parallel/AbstractMultiPhaseParallelIndexingTest.java:179]: Sure. this is the actual task status though.

**Pull request merged**